packages/batch-submitter/src/exec/run-batch-submitter.ts

```diff
@@ -320,4 +316,12 @@ export const run = async () => {
   if (requiredEnvVars.RUN_STATE_BATCH_SUBMITTER === 'true') {
     loop(() => stateBatchSubmitter.submitNextBatch())
   }
+
+  if (RUN_METRICS_SERVER === 'true') {
+    // Initialize metrics server
+    const metricsServer = createMetricsServer({
```

Comment:
Why store as a local variable, seems unused?

Reply:
There is no top level class that it can be assigned to, ideally it would be:

```js
this.metricsServer = createMetricsServer({})
```

or better yet

```js
this.metricsServer = new MetricsServer({})
```

It shouldn't be garbage collected if its not assigned to a variable, the variable kind of does nothing but I suppose it looks kind of weird without it. This is the reason why I like the `start()` and `stop()` API that I suggested in https://github.com/ethereum-optimism/optimism/pull/792/files#r627706069